File: app/src/main/java/com/elegion/android/ui/splash/SplashPresenter.java
package com.elegion.android.ui.splash;

import android.text.TextUtils;

import com.arellomobile.mvp.InjectViewState;
import com.elegion.android.data.Repository;
import com.elegion.android.ui.base.presenter.BasePresenter;

/**
 * @author mikhail.barannikov on 04.08.2017
 */
@InjectViewState
public class SplashPresenter extends BasePresenter<SplashView> {

    private final Repository mRepository;

    public SplashPresenter(Repository repository) {
        mRepository = repository;
    }

    public void timerFinish() {
        final String loginToken = mRepository.getLoginToken();
        if (TextUtils.isEmpty(loginToken)) {
            getViewState().openLogin();
        } else {
            getViewState().openFeatures();
        }
    }
}

File: app/src/main/kotlin/com/elegion/android/bankex/ui/splash/SplashPresenter.kt
package com.elegion.android.bankex.ui.splash

import com.arellomobile.mvp.InjectViewState
import com.elegion.android.bankex.data.Repository
import com.elegion.android.bankex.ui.base.presenter.BasePresenter

@InjectViewState
internal class SplashPresenter(private val repository: Repository) : BasePresenter<SplashView>() {

    fun timerFinish() {
        val onBoardingFlag = repository.onBoardingFlag
        if (onBoardingFlag) {
            viewState.openOnBoarding()
        } else {
            viewState.openWallet()
        }
    }
}
